SF iOS/SF iOS/Helpers/UserNotification/UNNotificationAttachment+EventNotifications.m

```diff
@@ -0,0 +1,41 @@
+//
+//  UNNotificationAttachment+EventNotifications.m
+//  Coffup
+//
+//  Created by Roderic Campbell on 5/1/19.
+//
+
+#import "UNNotificationAttachment+EventNotifications.h"
+
+@implementation UNNotificationAttachment (EventNotifications)
+
++ (UNNotificationAttachment *)withIdentifier:(NSString *)string URL:(NSURL *)url {
+    NSFileManager *fileManager = [NSFileManager defaultManager];
+    NSString *subFolderName = NSProcessInfo.processInfo.globallyUniqueString;
+
+    NSURL *subFolderURL = [[NSURL fileURLWithPath:NSTemporaryDirectory()] URLByAppendingPathComponent:subFolderName isDirectory:YES];
+
+    NSError *error = nil;
+    [fileManager createDirectoryAtURL:subFolderURL
+          withIntermediateDirectories:TRUE
+                           attributes:nil
+                                error:&error];
+    if (error != nil) {
+        NSLog(@"error is %@", error);
+        return nil;
+    }
+    NSString *imageFileIdentifier = [string stringByAppendingString:@".png"];
+    NSURL *fileURL = [subFolderURL URLByAppendingPathComponent:imageFileIdentifier];
+
+    // should probably be backgrounded
+    NSData *imageData = [NSData dataWithContentsOfURL:url];
+    if (![imageData writeToURL:fileURL atomically:TRUE]) {
+        return nil;
+    }
```

Comment:
@mcritz any thoughts on this guy? Think it's reasonable to go with this for now or does working in your infra from a few days/weeks ago make sense?

Reply:
Basically I save it to `NSTemporaryDirectory()` but there is a scenario where we actually use your cache and the app while running essentially never needs to wait for images (if they've gotten the push notification)

Reply:
I guess the issue here is that I need the image's local file URL for the notifications. That doesn't quite jive with the NSCache mechanism as far as I can tell.

Reply:
I'm going to pass on that update for now. Seems like there is some main thread stuff happening in there. I don't really want to mess with that for now.